abi/abi.go

```diff
@@ -87,6 +88,79 @@ func NewABI(s string) (*ABI, error) {
 	return NewABIFromReader(bytes.NewReader([]byte(s)))
 }
 
+func NewABIFromSlice(abi []compiler.AbiField) (*ABI, error) {
+	a := &ABI{}
+
+	for _, field := range abi {
+		switch field.Type {
+		case "constructor":
+			if a.Constructor != nil {
+				return nil, fmt.Errorf("multiple constructor declaration")
+			}
+			input, err := NewTupleTypeFromFields(field.Inputs)
+			if err != nil {
+				panic(err)
```

Comment:
Can we return an error instead of panicing?

Reply:
All panicking was from original code, so I'll update it to error propagation